lib/cocoapods/installer.rb

```diff
@@ -399,6 +400,21 @@ def verify_no_duplicate_framework_names
       end
     end
 
+    def verify_no_duplicate_library_names
```

Comment:
Good question. I don't generally like to add layers of abstraction for fewer than three uses (especially on code that doesn't change often), but we could do something along the lines of:

```ruby
def verify_no_duplicate_frameworks_and_libraries_names
  aggregate_targets.each do |aggregate_target|
    aggregate_target.user_build_configurations.keys.each do |config|
      pod_targets = aggregate_target.pod_targets_for_build_configuration(config)
      verify_no_duplicate_framework_names_in_pod_targets(targets)
      verify_no_duplicate_library_names_in_pod_targets(targets)
    end
  end
end

def verify_no_duplicate_framework_names_in_pod_targets(targets)
  vendored_frameworks = pod_targets.flat_map(&:file_accessors).flat_map(&:vendored_frameworks).uniq
  frameworks = vendored_frameworks.map { |fw| fw.basename('.framework') }
  frameworks += pod_targets.select { |pt| pt.should_build? && pt.requires_frameworks? }.map(&:product_module_name)

  duplicates = frameworks.group_by { |f| f }.select { |_, v| v.size > 1 }.keys
  unless duplicates.empty?
    raise Informative, "The '#{aggregate_target.label}' target has " \
      "frameworks with conflicting names: #{duplicates.to_sentence}."
  end
end

def verify_no_duplicate_library_names_in_pod_targets(targets)
  libraries = pod_targets.flat_map(&:file_accessors).flat_map(&:vendored_libraries).uniq.map(&:basename)
  libraries += pod_targets.select { |pt| pt.should_build? && !pt.requires_frameworks? }.map(&:product_name)
  duplicates = libraries.group_by { |l| l.to_s.downcase }.select { |_, v| v.size > 1 }.keys
  unless duplicates.empty?
    raise Informative, "The '#{aggregate_target.label}' target has " \
      "libraries with conflicting names: #{duplicates.to_sentence}."
  end
end
```

but I'm not sure that we actually get too much from it